### PR TITLE
Handle the exception instead of the old FALSE.

### PR DESCRIPTION
**Pull request by jordandukart**

### CODE REVIEW

File: includes/datastream.inc [hunk 1]
```diff
@@ -145,7 +145,13 @@ function islandora_rest_datastream_post_response(array $parameters) {
...
   $datastream->setContentFromFile($file['tmp_name']);
-  if (!$object->ingestDatastream($datastream)) {
+  // XXX: The return of FALSE will be deprecated in 7.x-1.10, remove this after.
+  try {
+    if (!$object->ingestDatastream($datastream)) {
+      throw new Exception('Conflict: Datastream already exists', 409);
+    }
+  }
+  catch (DatastreamExistsException $e) {
     throw new Exception('Conflict: Datastream already exists', 409);
```
Comment (adam-vessey) [inline includes/datastream.inc:155]: Should probably pass the exception as the third parameter here.

File: includes/datastream.inc [hunk 2]
```diff
@@ -145,7 +145,13 @@ function islandora_rest_datastream_post_response(array $parameters) {
   $datastream->mimetype = isset($request['mimeType']) ? $request['mimeType'] : $mime_detect->getMimetype($file['name']);
   $datastream->checksumType = isset($request['checksumType']) ? $request['checksumType'] : variable_get('islandora_checksum_checksum_type', 'DISABLED');
   $datastream->setContentFromFile($file['tmp_name']);
-  if (!$object->ingestDatastream($datastream)) {
+  // XXX: The return of FALSE will be deprecated in 7.x-1.10, remove this after.
+  try {
+    if (!$object->ingestDatastream($datastream)) {
+      throw new Exception('Conflict: Datastream already exists', 409);
```
Comment (adam-vessey) [inline includes/datastream.inc:151]: Deduping... could just throw a `DatastreamExistsException`, yeah?

**Review by adam-vessey — COMMENTED** [review]
Nothing major.

**Comment by jordandukart** [comment]
Git's acting weird and not seeing the new commit I pushed up, going to close and re-open this.

**Pull request closed without merging**